HomeWork/Home1/Main.java

```diff
@@ -0,0 +1,95 @@
+import java.util.Scanner;
+
+public class Main {
+    
+    // Function to find the largest negative value from a list of numbers
+    public static double findLargestNegative(double[] numbers) {
+        double largestNegative = Double.NEGATIVE_INFINITY;
+        
+        for (double num : numbers) {
+            if (num < 0 && num > largestNegative) {
+                largestNegative = num;
+            }
+        }
+        
+        return largestNegative != Double.NEGATIVE_INFINITY ? largestNegative : Double.NaN;
+    }
+    
+    // Function to calculate the roots of a quadratic equation
+    public static void calculateQuadraticRoots(double a, double b, double c) {
+        double discriminant = b * b - 4 * a * c;
+
+        if (discriminant > 0) {
+            double root1 = (-b + Math.sqrt(discriminant)) / (2 * a);
```

Comment:
we can optimize this part of code with precalculation of 1/(2*a) and Math.sqrt(discriminant) to avoid dupliication of that calls